src/Documents/DocumentHandler.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Documents;
+
+use App\Entity\Document;
+use League\Flysystem\FilesystemInterface;
+use Ramsey\Uuid\Uuid;
+
+class DocumentHandler
+{
+    public function __construct(private readonly FilesystemInterface $storage)
+    {
+    }
+
+    public function handleFile(Document $document): void
+    {
+        if (!$file = $document->getFile()) {
+            return;
+        }
+
+        $this->removeFile($document);
+
+        $document->setFilePath($path = sprintf(
+            '%s/%s.%s',
+            'files/documents/admin',
```

Comment:
IMO we don't need this detail in the file path, if we open this feature to simple user (not admin) we don't have to change this path.

```suggestion
            'files/documents',
```